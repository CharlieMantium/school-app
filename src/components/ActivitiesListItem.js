import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { EDIT_ACTIVITY_ROUTE } from '../constants/routes';

const ActivitiesListItem = ({ name, day, classNo, teacher, id }) => (
  <div>
    <Link to={EDIT_ACTIVITY_ROUTE(id)}>
      <h3>{name}</h3>
    </Link>
    <p>{`On ${day} on ${classNo} class with ${teacher}`}</p>
  </div>
);

ActivitiesListItem.propTypes = {
  name: PropTypes.string,
  day: PropTypes.string,
  classNo: PropTypes.string,
  teacher: PropTypes.string,
  id: PropTypes.string
};

export default ActivitiesListItem;
