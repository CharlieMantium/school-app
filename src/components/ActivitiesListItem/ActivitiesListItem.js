import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { FormattedMessage } from 'react-intl';
import uuid4 from 'uuid-v4';

import { EDIT_ACTIVITY_ROUTE } from 'constants/routes';
import activityPropTypeShape from 'prop-types/activity';
import { colors, effects, spacing, fontSizes } from 'styles/base';
import { divideByFilteredString } from 'helpers/divideByFilteredString';

const ActivityOrdinalWrapper = styled.p`
  display: inline;
  margin-right: ${spacing.sSize};
`;

const EditActivityLink = styled(Link)`
  color: ${colors.green};
  cursor: pointer;
  font-size: ${fontSizes.lFontSize};
  text-decoration: none;
  text-shadow: ${effects.outline(colors.black)};
  word-break: break-word;
`;

const ActivityDesc = styled.p`
  margin: ${spacing.xsSize} 0;
  word-break: break-word;
`;

const HighlightedSpan = styled.span`
  color: ${props => props.isHighlighted && `${colors.pink}`};
  background-color: ${props => props.isHighlighted && `${colors.turquoise}`};
`;

const ActivitiesListItem = ({ activity: { name, activityOrdinal, teacher, id, room }, filter }) => {
  const renderedValues = value => {
    if (filter) {
      return divideByFilteredString(value, filter).map(item => (
        <HighlightedSpan isHighlighted={item.isSearched} key={uuid4()}>
          {item.str}
        </HighlightedSpan>
      ));
    }
    return value;
  };
  return (
    <>
      <ActivityOrdinalWrapper>{`${activityOrdinal}.`}</ActivityOrdinalWrapper>
      <EditActivityLink data-test="react-link" to={EDIT_ACTIVITY_ROUTE(id)}>
        {renderedValues(name)}
      </EditActivityLink>
      <ActivityDesc>
        <FormattedMessage
          id="planView.activityDetails"
          defaultMessage="Teacher: {teacher}, in room: {room}"
          values={{ teacher: renderedValues(teacher), room: renderedValues(room) }}
          data-test="activity-details"
        />
      </ActivityDesc>
    </>
  );
};

ActivitiesListItem.propTypes = {
  activity: PropTypes.shape(activityPropTypeShape),
  filter: PropTypes.string,
};

ActivitiesListItem.defaultProps = {
  activity: {},
  filter: '',
};

const mapStateToProps = ({ filter }) => ({
  filter,
});

export { ActivitiesListItem as ActivitiesListItemUnwrapped };
export default connect(mapStateToProps)(ActivitiesListItem);
