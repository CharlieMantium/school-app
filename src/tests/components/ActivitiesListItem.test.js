import React from 'react';
import { shallow } from 'enzyme';

import ActivitiesListItem from '../../components/ActivitiesListItem';
import testState from '../fixtures/state';

describe('ActivitiesListItem', () => {
  it('should render ActivitiesListItem correctly', () => {
    const wrapper = shallow(<ActivitiesListItem activity={testState.activities.items[3]} />);
    const { name, day, classNo, teacher, room } = testState.activities.items[3];
    expect(wrapper.find('[data-test="react-link"]').exists()).toBe(true);
    expect(wrapper.find('[data-test="activity-name"]').text()).toBe(name);
    expect(wrapper.find('[data-test="activity-details"]').text()).toBe(
      `On ${day} on ${classNo} class with ${teacher} in room: ${room}`,
    );
  });
});
