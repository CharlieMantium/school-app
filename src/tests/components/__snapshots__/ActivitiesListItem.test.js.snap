// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivitiesListItem should render ActivitiesListItem correctly 1`] = `
<div>
  <Link
    to="/edit/4"
  >
    <h3>
      w-f
    </h3>
  </Link>
  <p>
    On wednesday on 7 class with RAMBO in room: gym
  </p>
</div>
`;
